tests/Integration/Behaviour/Features/Context/Domain/TitleFeatureContext.php

```diff
@@ -28,14 +28,109 @@
 
 namespace Tests\Integration\Behaviour\Features\Context\Domain;
 
+use Behat\Gherkin\Node\TableNode;
+use Language;
+use PHPUnit\Framework\Assert as Assert;
+use PrestaShop\PrestaShop\Core\Domain\Title\Command\AddTitleCommand;
 use PrestaShop\PrestaShop\Core\Domain\Title\Command\BulkDeleteTitleCommand;
 use PrestaShop\PrestaShop\Core\Domain\Title\Command\DeleteTitleCommand;
+use PrestaShop\PrestaShop\Core\Domain\Title\Command\EditTitleCommand;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\CannotAddTitleException;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\CannotUpdateTitleException;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleException;
 use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Title\Query\GetTitleForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Title\QueryResult\EditableTitle;
+use PrestaShop\PrestaShop\Core\Domain\Title\ValueObject\Gender;
+use PrestaShop\PrestaShop\Core\Domain\Title\ValueObject\TitleId;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
 use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+use Tests\Integration\Behaviour\Features\Context\Util\NoExceptionAlthoughExpectedException;
 use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
 
 class TitleFeatureContext extends AbstractDomainFeatureContext
 {
+    /**
+     * @var int default lang id from configs
+     */
+    protected $defaultLangId;
+
+    public function __construct()
+    {
+        $configuration = CommonFeatureContext::getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->defaultLangId = (int) $configuration->get('PS_LANG_DEFAULT');
+    }
+
+    /**
+     * @When I add a new title :titleReference with the following properties:
+     *
+     * @param string $titleReference
+     * @param TableNode $table
+     */
+    public function createTitle(string $titleReference, TableNode $table): void
+    {
+        $data = $this->localizeByRows($table);
+
+        /** @var FormChoiceProviderInterface $provider */
+        $provider = CommonFeatureContext::getContainer()->get('prestashop.core.form.choice_provider.gender_choice_provider');
+        $availableGendersTypes = $provider->getChoices();
+        if (!isset($availableGendersTypes[$data['type']])) {
+            throw new RuntimeException(sprintf('Gender type "%s" is not available.', $data['type']));
+        }
+
+        try {
+            /** @var TitleId $titleId */
+            $titleId = $this->getCommandBus()->handle(new AddTitleCommand(
+                $data['name'],
+                new Gender((int) $availableGendersTypes[$data['type']]),
+                null,
+                null,
+                null
```

Comment:
If these three parameters are not mandatory maybe they should be handled by a dedicated setter (for the three parameters), or at least the should be nullable in the constructor so we don't need to always set null explicitely

Reply:
Fixed